ppb/engine.py

```diff
@@ -33,7 +35,7 @@ def __init__(self, first_scene: Type, *,
         # Engine State
         self.scenes = []
         self.events = deque()
-        self.event_extensions = defaultdict(dict)
+        self.event_extensions: DefaultDict[type, Dict[str, Callable[[], Any]]] = defaultdict(dict)
```

Comment:
I don't think the type annotation is useful (it should be at the class level for `mypy` to be able to do something with it.)

Reply:
I put it in because my linter was whining about types down in the publish function.

Reply:
OK, that's weird.